### ISSUE TITLE
Fetching from git fails

### ISSUE THREAD

**Issue by mmuman** [haikuports/haikuporter#182]
It seems the code disagrees on where the sources have been checked out. git puts them in `download/Ham` while the unpack code expects it to be in a folder named after the hash. Same for other git+* recipes.

```
LANG=C haikuporter --get-dependencies ham_x86
Checking if any dependency-infos need to be updated ...
Looking for stale dependency-infos ...
----------------------------------------------------------------------
sys-devel::ham_x86-0.1
        /work/haikuports/haikuports/sys-devel/ham/ham-0.1.recipe
----------------------------------------------------------------------

Downloading: git+https://github.com/weinhold/Ham#b83d767047a93ed58717a6c7662b8838342a54fb ...
Warning: UNSAFE SOURCES ARE BAD AND SHOULD NOT BE USED IN PRODUCTION
Warning: PLEASE MOVE TO A STATIC ARCHIVE DOWNLOAD WITH CHECKSUM ASAP!
Clonage dans le dépôt nu '/work/haikuports/haikuports/sys-devel/ham/download/Ham'
Unpacking source of Ham
tar: /work/haikuports/haikuports/sys-devel/ham/work-x86-0.1/sources/ham-0.1: Cannot open: No such file or directory
tar: Error is not recoverable: exiting now
Command 'git archive b83d767047a93ed58717a6c7662b8838342a54fb | tar -x -C "/work/haikuports/haikuports/sys-devel/ham/work-x86-0.1/sources/ham-0.1"' returned non-zero exit status 2
```

**Comment by phw** [comment]
A partial workaround seems to be setting `SOURCE_DIR="."`. This allowed me to build the package, but the it fails building the source package.

**Comment by phw** [comment]
A partial workaround seems to be setting `SOURCE_DIR="."`, this allows building the binary package. Unfortunately this then causes a problem building the source package, so you have to run haikuporter with the `--no-source-packages` flag.

**Comment by diversys** [comment]
`SOURCE_URI` in ham recipe should be changed to https only and pint to the latest commit archive. See `srcGitRev` in other recipes.

**Comment by pulkomandy** [comment]
Yes, but if we don't plan to maintain support for fetching from git we should remove that completely from haikuports and not keep unused and broken code around. Or if we decide it's important to keep the feature, we should fix it.

**Comment by mmuman** [comment]
Well it might still be useful for CI purposes (making sure HEAD builds from time to time)…

**Comment by phw** [comment]
I think it is a useful feature, especially during development. I maintain a development package recipe for my application, which allows me to always ensure the packaging stays up-to-date. When there is an actual release I just have to update the SOURCE_URI + checksum for the archive. I also considered setting up a CI build for having always a devlopment hpkg available, but have not yet done so.

It would be totally reasonable to disallow fetching from git for packages in haikuports. But even for production use there are some cases where git fetching (with commit pinning) makes things easier to handle.
